upstream-sha256: d0c00c66cc03f4f8071621b28e9bae8e8749fd51f51047def12864fc8c4ec45d
actual-sha256: e11e122b18dca53cb362340d80e12ea446aacef6dcee543c0924400a8a82de75
--- upstream.expected
+++ tolk.actual
@@ -18,7 +18,7 @@
   float val4 = (*(data4_16+(lidx1+4)));
   float val5 = (*(data4_16+(lidx1+8)));
   float4 val6 = (*((float4*)((data3_16+alu0))));
-  *(data0_16+(lidx1+alu0)) = ((*(buf0+0))+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3));
+  *(data0_16+(lidx1+alu0)) = ((*(buf0+0))+(0.0f+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3)));
 }
 ---
 typedef unsigned int uint;
@@ -41,7 +41,7 @@
   float val4 = (*(data4_16+(lidx1+4)));
   float val5 = (*(data4_16+(lidx1+8)));
   float4 val6 = (*((float4*)((data3_16+alu0))));
-  *(data0_16+(lidx1+alu0)) = (2.0f*((*(buf0+0))+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3)));
+  *(data0_16+(lidx1+alu0)) = (2.0f*((*(buf0+0))+(0.0f+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3))));
 }
 ---
 typedef unsigned int uint;
@@ -56,7 +56,7 @@
   float val0 = (*(data1_16+alu1));
   float4 val1 = (*((float4*)((data2_16+alu0))));
   float4 val2 = (*((float4*)((data3_16+(lidx1<<2)))));
-  *(data0_16+alu1) = ((2.0f*val0)+(val1.x*val2.x)+(val1.y*val2.y)+(val1.z*val2.z)+(val1.w*val2.w));
+  *(data0_16+alu1) = ((2.0f*val0)+(0.0f+(val1.x*val2.x)+(val1.y*val2.y)+(val1.z*val2.z)+(val1.w*val2.w)));
 }
 ---
 typedef unsigned int uint;
@@ -82,7 +82,7 @@
   float val7 = (*(data4_16+(lidx1+12)));
   float val8 = (*(data4_16+(lidx1+4)));
   float val9 = (*(data4_16+(lidx1+8)));
-  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7));
+  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(0.0f+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7)));
 }
 ---
 typedef unsigned int uint;
@@ -108,7 +108,7 @@
   float val7 = (*(data4_16+(lidx1+12)));
   float val8 = (*(data4_16+(lidx1+4)));
   float val9 = (*(data4_16+(lidx1+8)));
-  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7));
+  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(0.0f+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7)));
 }
 ---
 typedef unsigned int uint;
@@ -121,5 +121,5 @@
   float4 val0 = (*((float4*)((data1_16+alu0))));
   int lidx1 = threadIdx.y; /* 4 */
   float4 val1 = (*((float4*)((data2_16+(lidx1<<2)))));
-  *(data0_16+(lidx1+alu0)) = ((val0.x*val1.x)+(val0.y*val1.y)+(val0.z*val1.z)+(val0.w*val1.w));
+  *(data0_16+(lidx1+alu0)) = (0.0f+(val0.x*val1.x)+(val0.y*val1.y)+(val0.z*val1.z)+(val0.w*val1.w));
 }
